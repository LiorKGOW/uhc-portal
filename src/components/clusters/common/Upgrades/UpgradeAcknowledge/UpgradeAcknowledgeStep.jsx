import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import {
  Alert,
  Divider,
  Form,
  FormGroup,
  Grid,
  GridItem,
  TextInput,
  Title,
} from '@patternfly/react-core';

import ExternalLink from '../../../../common/ExternalLink';

import './UpgradeAcknowledgeStep.scss';

const ackWord = 'Acknowledge';

const UpgradeAcknowledgeStep = (props) => {
  const [typedWord, setTypedWord] = useState('');

  const { initiallyConfirmed, confirmed, fromVersion, toVersion, unmetAcknowledgements } = props;

  useEffect(() => {
    if (initiallyConfirmed) {
      setTypedWord(ackWord);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    if (typedWord === ackWord) {
      return confirmed(true);
    }
    return confirmed(false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [typedWord]);

  return (
    <Grid hasGutter className="ocm-upgrade-ack-step">
      <GridItem>
        <Title className="wizard-step-title" size="lg" headingLevel="h3">
          1. Review and prevent update issues
        </Title>
        <ul className="wizard-step-body">
          {unmetAcknowledgements.map((ack) => (
            <li data-testid="unmetAcknowledgement">
              {ack.description ? <p>{ack.description}</p> : null}
              <Alert
                id="upgrade-ack-alert"
                isPlain
                isInline
                variant="warning"
                title={ack.warning_message}
              >
                {ack.documentation_url ? (
                  <ExternalLink href={ack.documentation_url}>Learn more</ExternalLink>
                ) : null}
              </Alert>
            </li>
          ))}
        </ul>
      </GridItem>

      <Divider />
      <GridItem>
        <Title className="wizard-step-title" size="lg" headingLevel="h3">
          2. Acknowledge
        </Title>
        <div className="wizard-step-body">
          <p>
            {`
              I acknowledge I have resolved any potential update issues on my cluster, 
              and the cluster is ready to update from OpenShift Container Platform
              from ${fromVersion} to ${toVersion}.
              `}
          </p>

          <Form
            onSubmit={(e) => {
              e.preventDefault();
            }}
          >
            <FormGroup
              label={
                <span className="ocm-upgrade-ack-step_text-input-label">
                  Confirm the above by typing <strong>{ackWord}</strong> below:
                </span>
              }
              isRequired
              fieldId="upgradeAcknowledgementModalText"
            >
              <TextInput
                value={typedWord}
                isRequired
                id="upgradeAcknowledgementModalText"
                type="text"
                onChange={(val) => setTypedWord(val)}
                placeholder="Type here"
                data-testid="acknowledgeTextInput"
              />
            </FormGroup>
          </Form>
        </div>
      </GridItem>
    </Grid>
  );
};

UpgradeAcknowledgeStep.propTypes = {
  confirmed: PropTypes.func, // returns a boolean
  unmetAcknowledgements: PropTypes.array,
  fromVersion: PropTypes.string,
  toVersion: PropTypes.string,
  initiallyConfirmed: PropTypes.bool,
};

UpgradeAcknowledgeStep.defaultProps = {
  initiallyConfirmed: false,
};
export default UpgradeAcknowledgeStep;
