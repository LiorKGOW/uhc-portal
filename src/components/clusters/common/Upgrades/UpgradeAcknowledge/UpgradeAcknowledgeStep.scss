.ocm-upgrade-ack-modal{
  // h2{
  //   font-weight: var(--pf-global--FontWeight--bold);
  //   font-size: var(--pf-global--FontSize--lg);
  // }
  
  p{
    font-size: var(--pf-global--FontSize--sm);
    margin-top: var(--pf-global--spacer--sm);
    margin-bottom: var(--pf-global--spacer--sm);
  }
}