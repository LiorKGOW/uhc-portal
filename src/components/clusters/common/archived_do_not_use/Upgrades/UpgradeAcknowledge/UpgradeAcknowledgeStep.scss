.ocm-upgrade-ack-step {
  p {
    font-size: var(--pf-v5-global--FontSize--sm);
    margin-top: var(--pf-v5-global--spacer--sm);
    margin-bottom: var(--pf-v5-global--spacer--sm);
  }
  .ocm-upgrade-ack-step_text-input-label {
    font-size: var(--pf-v5-global--FontSize--md);
    font-weight: var(--pf-v5-global--FontWeight--normal);
  }
}
