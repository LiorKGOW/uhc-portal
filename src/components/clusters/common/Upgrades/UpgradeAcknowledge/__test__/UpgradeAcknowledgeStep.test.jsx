import React from 'react';
import { mount } from 'enzyme';

import UpgradeAcknowledgeStep from '../UpgradeAcknowledgeStep';

describe('<UpgradeAcknowledgeStep>', () => {
  let wrapper;
  const ackWord = 'Acknowledge';
  const confirmedMock = jest.fn();

  const ackArray = [
    { description: 'my upgrade gate', warning_message: 'my ack warning message', documentation_url: 'my doc url' },
    { description: 'my other upgrade gate', warning_message: 'my other ack warning message', documentation_url: 'my other doc url' },
  ];

  beforeEach(() => {
    confirmedMock.mockClear();

    wrapper = mount(
      <UpgradeAcknowledgeStep
        confirmed={confirmedMock}
        unmetAcknowledgements={ackArray}
        fromVersion="4.8.2"
        toVersion="4.9.12"
      />,
    );
  });

  it('should render correctly on load', () => {
    expect(wrapper.find('[data-testid="unmetAcknowledgement"]')).toHaveLength(ackArray.length);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not be confirmed on empty confirm text', () => {
    wrapper.find('TextInput[data-testid="acknowledgeTextInput"]').invoke('onChange')('');
    wrapper.update();
    expect(confirmedMock.mock.calls[confirmedMock.mock.calls.length - 1][0]).toEqual(false);
  });

  it('should not be confirmed if wrong confirm word is typed', () => {
    wrapper.find('TextInput[data-testid="acknowledgeTextInput"]').invoke('onChange')('notCorrectWord');
    wrapper.update();
    expect(confirmedMock.mock.calls[confirmedMock.mock.calls.length - 1][0]).toEqual(false);
  });

  it('should confirm if correct confirm word is typed', () => {
    wrapper.find('TextInput[data-testid="acknowledgeTextInput"]').invoke('onChange')(ackWord);
    wrapper.update();
    expect(confirmedMock.mock.calls[confirmedMock.mock.calls.length - 1][0]).toEqual(true);
  });

  it('is confirmed on load if initiallyConfirmed is true', () => {
    wrapper = mount(
      <UpgradeAcknowledgeStep
        confirmed={confirmedMock}
        unmetAcknowledgements={ackArray}
        fromVersion="4.8.2"
        toVersion="4.9.12"
        initiallyConfirmed
      />,
    );

    expect(confirmedMock.mock.calls[confirmedMock.mock.calls.length - 1][0]).toEqual(true);
  });
});
