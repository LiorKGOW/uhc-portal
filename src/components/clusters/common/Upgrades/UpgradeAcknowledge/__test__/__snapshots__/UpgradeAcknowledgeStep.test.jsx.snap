// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpgradeAcknowledgeStep> should render correctly on load 1`] = `
<UpgradeAcknowledgeStep
  confirmed={
    [MockFunction] {
      "calls": Array [
        Array [
          false,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  fromVersion="4.8.2"
  initiallyConfirmed={false}
  toVersion="4.9.12"
  unmetAcknowledgements={
    Array [
      Object {
        "description": "my upgrade gate",
        "documentation_url": "my doc url",
        "warning_message": "my ack warning message",
      },
      Object {
        "description": "my other upgrade gate",
        "documentation_url": "my other doc url",
        "warning_message": "my other ack warning message",
      },
    ]
  }
>
  <Grid
    hasGutter={true}
  >
    <div
      className="pf-l-grid pf-m-gutter"
    >
      <GridItem>
        <div
          className="pf-l-grid__item"
        >
          <Title
            className="wizard-step-title"
            headingLevel="h3"
            size="lg"
          >
            <h3
              className="pf-c-title pf-m-lg wizard-step-title"
              data-ouia-component-id="OUIA-Generated-Title-1"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe={true}
            >
              1. Review and prevent update issues
            </h3>
          </Title>
          <ul
            className="wizard-step-body"
          >
            <li
              data-testid="unmetAcknowledgement"
            >
              <p>
                my upgrade gate
              </p>
              <Alert
                id="upgrade-ack-alert"
                isInline={true}
                isPlain={true}
                title="my ack warning message"
                variant="warning"
              >
                <div
                  aria-label="Warning Alert"
                  className="pf-c-alert pf-m-inline pf-m-plain pf-m-warning"
                  data-ouia-component-id="OUIA-Generated-Alert-warning-1"
                  data-ouia-component-type="PF4/Alert"
                  data-ouia-safe={true}
                  id="upgrade-ack-alert"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <AlertIcon
                    variant="warning"
                  >
                    <div
                      className="pf-c-alert__icon"
                    >
                      <ExclamationTriangleIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                      >
                        <svg
                          aria-hidden={true}
                          aria-labelledby={null}
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style={
                            Object {
                              "verticalAlign": "-0.125em",
                            }
                          }
                          viewBox="0 0 576 512"
                          width="1em"
                        >
                          <path
                            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                          />
                        </svg>
                      </ExclamationTriangleIcon>
                    </div>
                  </AlertIcon>
                  <h4
                    className="pf-c-alert__title"
                  >
                    <span
                      className="pf-u-screen-reader"
                    >
                      Warning alert:
                    </span>
                    my ack warning message
                  </h4>
                  <div
                    className="pf-c-alert__description"
                  >
                    <ExternalLink
                      href="my doc url"
                    >
                      <a
                        href="my doc url"
                        rel="noreferrer noopener"
                        target="_blank"
                      >
                        Learn more
                         
                        <ExternalLinkAltIcon
                          color="#0066cc"
                          noVerticalAlign={false}
                          size="sm"
                        >
                          <svg
                            aria-hidden={true}
                            aria-labelledby={null}
                            fill="#0066cc"
                            height="1em"
                            role="img"
                            style={
                              Object {
                                "verticalAlign": "-0.125em",
                              }
                            }
                            viewBox="0 0 512 512"
                            width="1em"
                          >
                            <path
                              d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                            />
                          </svg>
                        </ExternalLinkAltIcon>
                      </a>
                    </ExternalLink>
                  </div>
                </div>
              </Alert>
            </li>
            <li
              data-testid="unmetAcknowledgement"
            >
              <p>
                my other upgrade gate
              </p>
              <Alert
                id="upgrade-ack-alert"
                isInline={true}
                isPlain={true}
                title="my other ack warning message"
                variant="warning"
              >
                <div
                  aria-label="Warning Alert"
                  className="pf-c-alert pf-m-inline pf-m-plain pf-m-warning"
                  data-ouia-component-id="OUIA-Generated-Alert-warning-2"
                  data-ouia-component-type="PF4/Alert"
                  data-ouia-safe={true}
                  id="upgrade-ack-alert"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <AlertIcon
                    variant="warning"
                  >
                    <div
                      className="pf-c-alert__icon"
                    >
                      <ExclamationTriangleIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                      >
                        <svg
                          aria-hidden={true}
                          aria-labelledby={null}
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style={
                            Object {
                              "verticalAlign": "-0.125em",
                            }
                          }
                          viewBox="0 0 576 512"
                          width="1em"
                        >
                          <path
                            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                          />
                        </svg>
                      </ExclamationTriangleIcon>
                    </div>
                  </AlertIcon>
                  <h4
                    className="pf-c-alert__title"
                  >
                    <span
                      className="pf-u-screen-reader"
                    >
                      Warning alert:
                    </span>
                    my other ack warning message
                  </h4>
                  <div
                    className="pf-c-alert__description"
                  >
                    <ExternalLink
                      href="my other doc url"
                    >
                      <a
                        href="my other doc url"
                        rel="noreferrer noopener"
                        target="_blank"
                      >
                        Learn more
                         
                        <ExternalLinkAltIcon
                          color="#0066cc"
                          noVerticalAlign={false}
                          size="sm"
                        >
                          <svg
                            aria-hidden={true}
                            aria-labelledby={null}
                            fill="#0066cc"
                            height="1em"
                            role="img"
                            style={
                              Object {
                                "verticalAlign": "-0.125em",
                              }
                            }
                            viewBox="0 0 512 512"
                            width="1em"
                          >
                            <path
                              d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                            />
                          </svg>
                        </ExternalLinkAltIcon>
                      </a>
                    </ExternalLink>
                  </div>
                </div>
              </Alert>
            </li>
          </ul>
        </div>
      </GridItem>
      <Divider>
        <hr
          className="pf-c-divider"
        />
      </Divider>
      <GridItem>
        <div
          className="pf-l-grid__item"
        >
          <Title
            className="wizard-step-title"
            headingLevel="h3"
            size="lg"
          >
            <h3
              className="pf-c-title pf-m-lg wizard-step-title"
              data-ouia-component-id="OUIA-Generated-Title-2"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe={true}
            >
              2. Acknowledge
            </h3>
          </Title>
          <div
            className="wizard-step-body"
          >
            <p>
              
              I acknowledge I have resolved any potential update issues on my cluster, 
              and the cluster is ready to update from OpenShift Container Platform
              from 4.8.2 to 4.9.12.
              
            </p>
            <Form
              onSubmit={[Function]}
            >
              <form
                className="pf-c-form"
                noValidate={true}
                onSubmit={[Function]}
              >
                <FormGroup
                  fieldId="upgradeAcknowledgementModalText"
                  isRequired={true}
                  label={
                    <span
                      className="ocm-upgrade-ack-modal_text-input-label"
                    >
                      Confirm the above by typing
                       
                      <strong>
                        Acknowledge
                      </strong>
                       
                      below:
                    </span>
                  }
                >
                  <div
                    className="pf-c-form__group"
                  >
                    <div
                      className="pf-c-form__group-label"
                    >
                      <label
                        className="pf-c-form__label"
                        htmlFor="upgradeAcknowledgementModalText"
                      >
                        <span
                          className="pf-c-form__label-text"
                        >
                          <span
                            className="ocm-upgrade-ack-modal_text-input-label"
                          >
                            Confirm the above by typing
                             
                            <strong>
                              Acknowledge
                            </strong>
                             
                            below:
                          </span>
                        </span>
                        <span
                          aria-hidden="true"
                          className="pf-c-form__label-required"
                        >
                           
                          *
                        </span>
                      </label>
                       
                    </div>
                    <div
                      className="pf-c-form__group-control"
                    >
                      <TextInput
                        data-testid="acknowledgeTextInput"
                        id="upgradeAcknowledgementModalText"
                        isRequired={true}
                        onChange={[Function]}
                        placeholder="Type here"
                        type="text"
                        value=""
                      >
                        <TextInputBase
                          aria-label={null}
                          className=""
                          data-testid="acknowledgeTextInput"
                          id="upgradeAcknowledgementModalText"
                          innerRef={null}
                          isDisabled={false}
                          isLeftTruncated={false}
                          isReadOnly={false}
                          isRequired={true}
                          onChange={[Function]}
                          ouiaSafe={true}
                          placeholder="Type here"
                          type="text"
                          validated="default"
                          value=""
                        >
                          <input
                            aria-invalid={false}
                            aria-label={null}
                            className="pf-c-form-control"
                            data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                            data-ouia-component-type="PF4/TextInput"
                            data-ouia-safe={true}
                            data-testid="acknowledgeTextInput"
                            disabled={false}
                            id="upgradeAcknowledgementModalText"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            placeholder="Type here"
                            readOnly={false}
                            required={true}
                            type="text"
                            value=""
                          />
                        </TextInputBase>
                      </TextInput>
                    </div>
                  </div>
                </FormGroup>
              </form>
            </Form>
          </div>
        </div>
      </GridItem>
    </div>
  </Grid>
</UpgradeAcknowledgeStep>
`;
